flink-runtime/src/test/java/org/apache/flink/runtime/operators/AbstractOuterJoinTaskTest.java

```diff
@@ -0,0 +1,501 @@
+				getOuterJoinDriver();
+
+		addInput(new UniformIntTupleGenerator(keyCnt1, valCnt1, true), serializer);
+		addInput(new UniformIntTupleGenerator(keyCnt2, valCnt2, true), serializer);
+
+		try {
+			testDriver(testTask, MockJoinStub.class);
+		} catch (Exception e) {
+			e.printStackTrace();
+			Assert.fail("The test caused an exception.");
+		}
+
+		final int expCnt = calculateExpectedCount(keyCnt1, valCnt1, keyCnt2, valCnt2);
+
+		Assert.assertTrue("Resultset size was "+this.outList.size()+". Expected was "+expCnt, this.outList.size() == expCnt);
+
+		this.outList.clear();
+	}
+
+	@Test
+	public void testFailingOuterJoinTask() {
+		int keyCnt1 = 20;
+		int valCnt1 = 20;
+
+		int keyCnt2 = 20;
+		int valCnt2 = 20;
+
+		setOutput(new DiscardingOutputCollector<Tuple2<Integer, Integer>>());
+		addDriverComparator(this.comparator1);
+		addDriverComparator(this.comparator2);
+		getTaskConfig().setDriverPairComparator(new RuntimePairComparatorFactory());
+		getTaskConfig().setDriverStrategy(driverStrategy);
+		getTaskConfig().setRelativeMemoryDriver(bnljn_frac);
+		setNumFileHandlesForSort(4);
+
+		final AbstractOuterJoinDriver<Tuple2<Integer, Integer>, Tuple2<Integer, Integer>, Tuple2<Integer, Integer>> testTask =
+				getOuterJoinDriver();
+
+		addInput(new UniformIntTupleGenerator(keyCnt1, valCnt1, true), serializer);
+		addInput(new UniformIntTupleGenerator(keyCnt2, valCnt2, true), serializer);
+
+		try {
+			testDriver(testTask, MockFailingJoinStub.class);
+			Assert.fail("Driver did not forward Exception.");
+		} catch (ExpectedTestException e) {
+			// good!
+		} catch (Exception e) {
+			e.printStackTrace();
+			Assert.fail("The test caused an exception.");
+		}
+	}
+
+	@Test
+	public void testCancelOuterJoinTaskWhileSort1() {
+		int keyCnt = 20;
+		int valCnt = 20;
+
+		setOutput(new DiscardingOutputCollector<Tuple2<Integer, Integer>>());
+		addDriverComparator(this.comparator1);
+		addDriverComparator(this.comparator2);
+		getTaskConfig().setDriverPairComparator(new RuntimePairComparatorFactory());
+		getTaskConfig().setDriverStrategy(driverStrategy);
+		getTaskConfig().setRelativeMemoryDriver(bnljn_frac);
+		setNumFileHandlesForSort(4);
+
+		final AbstractOuterJoinDriver<Tuple2<Integer, Integer>, Tuple2<Integer, Integer>, Tuple2<Integer, Integer>> testTask =
+				getOuterJoinDriver();
+
+		try {
+			addInputSorted(new DelayingIterator<>(new InfiniteIntTupleIterator(), 1), this.serializer, this.comparator1.duplicate());
+			addInput(new UniformIntTupleGenerator(keyCnt, valCnt, true), serializer);
+		} catch (Exception e) {
+			e.printStackTrace();
+			Assert.fail("The test caused an exception.");
+		}
+
+		final AtomicBoolean success = new AtomicBoolean(false);
+
+		Thread taskRunner = new Thread() {
+			@Override
+			public void run() {
+				try {
+					testDriver(testTask, MockJoinStub.class);
+					success.set(true);
+				} catch (Exception ie) {
+					ie.printStackTrace();
```

Comment:
Shall such an exception fail the test? If so, then you have to handle it appropriately.